test/integration/test-fapi.h

```diff
@@ -5,13 +5,127 @@
+   { \
+        json_object *jso = NULL; \
+        json_object *jso1 = NULL; \
+        json_object *jso2 = NULL; \
+        size_t i, n; \
+        n = sizeof(FIELD_LIST) / sizeof(FIELD_LIST[0]); \
+        jso = json_tokener_parse(JSON_STRING); \
+        if (!jso) { \
+            LOG_ERROR("Invalid JSON"); \
+            goto error; \
+        } \
+        jso1 = jso; \
+        for (i = 0; i < n; i++) { \
+            if (strlen(FIELD_LIST[i]) == 1 && \
+                FIELD_LIST[i][0] >= '0' && FIELD_LIST[i][0] <= '9') { \
+                jso2 = json_object_array_get_idx(jso1, FIELD_LIST[i][0]  - '0'); \
+                ASSERT(jso2); \
+            } \
+            else if (!jso1 || !json_object_object_get_ex(jso1, FIELD_LIST[i],  &jso2)) { \
+                json_object_put(jso); \
+                LOG_ERROR("%s not found.", FIELD_LIST[i]); \
+                goto error; \
+            } \
+            jso1 = jso2; \
+        } \
+        if (strlen(SUBSTRING) > 0 && !strstr( json_object_get_string(jso1), SUBSTRING)) { \
+            json_object_put(jso); \
+            LOG_ERROR("Sub string %s not found.", SUBSTRING); \
+            goto error; \
+        } \
+        json_object_put(jso); \
+    }
+
+/* It will be checked whether two json objects are equal. */
+#define CHECK_JSON(JSON1, JSON2, LABEL) \
+        { \
+            json_object *jso1 = NULL; \
+            json_object *jso2 = NULL; \
+            jso1 = json_tokener_parse(JSON1) ; \
+            ASSERT(jso1) ; \
+            if (!jso1) { \
+                LOG_ERROR("Invalid JSON") ;\
+                goto LABEL ;\
+            }                                        \
+            jso2 = json_tokener_parse(JSON2) ;\
+            ASSERT(jso2) ;\
+            if (!jso2) { \
+                LOG_ERROR("Invalid JSON") ;\
+                goto LABEL ;\
+            }                                        \
+            if (!cmp_jso(jso1, jso2)) { \
+                json_object_put(jso1) ; \
+                json_object_put(jso2) ; \
+                goto LABEL; \
+            } \
+            json_object_put(jso1); \
+            json_object_put(jso2); \
+        }
+
+/* It will be checked whether a json object is included in a list
+   of json objects. */
+#define CHECK_JSON_LIST(LIST, JSO_STRING, LABEL) \
+    { \
+        size_t i, n; \
+        n = sizeof(LIST) / sizeof(LIST[0]); \
+        json_object *jso1 = json_tokener_parse(JSO_STRING); \
+        ASSERT(jso1); \
+        json_object *jso2 = NULL; \
+        for (i = 0; i < n; i++) { \
+             jso2 = json_tokener_parse(LIST[i]); \
+             ASSERT(jso2); \
+             if (cmp_jso(jso1, jso2)) { \
+                break; \
+             } \
+             json_object_put(jso2); \
+        } \
+        if (i >=  n) { \
+            json_object_put(jso1);       \
+            json_object_put(jso2);          \
+            LOG_ERROR("Mismatch" ); \
```

Comment:
Unnecessary Whitespace

Reply:
@cplappert Thank you for the review.  I just pushed the changes.